tizen-api/src/tizen-api-pipeline.c

```diff
@@ -283,18 +308,57 @@ nns_pipeline_construct (const char *pipeline_description, nns_pipeline_h * pipe)
 int
 nns_pipeline_destroy (nns_pipeline_h pipe)
 {
-  /* nns_pipeline *p = pipe; */
+  nns_pipeline *p = pipe;
+  GstStateChangeReturn scret;
+  GstState state, pending;
+
+  if (p == NULL)
+    return NNS_ERROR_INVALID_PARAMETER;
+
+  g_mutex_lock (&p->lock);
+
+  /* Check if it's PLAYING */
+  scret = gst_element_get_state (p->element, &state, &pending, 10000000UL);     /* 10ms */
+  if (scret == GST_STATE_CHANGE_SUCCESS && state == GST_STATE_PLAYING) {
+    /* Pause the pipeline if it's Playing */
+    scret = gst_element_set_state (p->element, GST_STATE_PAUSED);
+    if (scret == GST_STATE_CHANGE_FAILURE) {
+      g_mutex_unlock (&p->lock);
+      return NNS_ERROR_PIPELINE_FAIL;
+    }
 
-  /** @todo NYI */
+    do {
+      /* timeount = 100 000 000 ns (100 ms) */
+      scret = gst_element_get_state (p->element, &state, &pending, 100000000UL);
+      if (scret == GST_STATE_CHANGE_FAILURE || scret == GST_STATE_CHANGE_ASYNC) {
+        g_mutex_unlock (&p->lock);
+        return NNS_ERROR_PIPELINE_FAIL;
+      }
+    } while (scret != GST_STATE_CHANGE_SUCCESS);
```

Comment:
This can be `GST_STATE_CHANGE_NOI_PREROLL` as well

Reply:
Nah... Let's simply kill everything after the first wait.